nvflare/app_common/aggregators/intime_accumulate_model_aggregator.py

```diff
@@ -55,38 +57,46 @@ def __init__(self, expected_data_kind=DataKind.WEIGHT_DIFF, exclude_vars=None, a
             self.expected_data_kind = {self._single_dxo_key: expected_data_kind}
 
         # Check exclude_vars
-        if isinstance(exclude_vars, dict):
-            exclude_vars_dict = dict()  # build default dict for expected entries not specified by the user
-            for k, v in exclude_vars.items():
-                if not isinstance(v, str):
-                    raise ValueError(f"exclude_vars[{k}] = {v} not a string but {type(v)}! Expected type regex string.")
-                if k not in self.expected_data_kind:
-                    exclude_vars_dict[k] = None
-                else:
-                    exclude_vars_dict[k] = v
-            self.exclude_vars = exclude_vars_dict
-        else:
-            self.exclude_vars = {self._single_dxo_key: exclude_vars}
+        exclude_vars_dict = dict()
+        for k in self.expected_data_kind.keys():
+            if isinstance(exclude_vars, dict):
+                if k in exclude_vars:
+                    if not isinstance(exclude_vars[k], str):
+                        raise ValueError(
+                            f"exclude_vars[{k}] = {exclude_vars[k]} not a string but {type(exclude_vars[k])}! "
+                            f"Expected type regex string."
+                        )
+                    exclude_vars_dict[k] = exclude_vars[k]
+            else:
+                exclude_vars_dict[k] = exclude_vars  # assume same exclude vars for each entry of DXO collection.
+        if self._single_dxo_key in self.expected_data_kind:
+            exclude_vars_dict[self._single_dxo_key] = exclude_vars
+        self.exclude_vars = exclude_vars_dict
 
         # Check aggregation weights
         aggregation_weights = aggregation_weights or {}
         aggregation_weights_dict = dict()
-        for key in self.expected_data_kind.keys():
-            if key in aggregation_weights:
+        for k in self.expected_data_kind.keys():
+            if k in aggregation_weights:
                 aggregation_weights_dict[k] = aggregation_weights[k]
             else:
-                aggregation_weights_dict[self._single_dxo_key] = aggregation_weights
+                aggregation_weights_dict[
+                    k
+                ] = aggregation_weights  # assume same aggregation weights for each entry of DXO collection.
```

Comment:
suggest move this comment from back to top.

```
# assume same aggregation weights for each entry of DXO collection.
aggregation_weights_dict[k] = aggregation_weights
```

Reply:
Good point. I changed it.